torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -825,5 +827,39 @@ def call_function(
         )
 
 
+class OutDtypeHigherOrderVariable(TorchHigherOrderOperatorVariable):
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        from . import ConstantVariable
+        from .builder import wrap_fx_proxy
+
+        p_args = tuple(arg.as_proxy() for arg in args)
+        op = p_args[0]
+        output_dtype = p_args[1]
+        real_sub_args = pytree.tree_map_only(
+            torch.fx.Proxy, lambda a: get_real_value(a.node, tx.output), p_args[2:]
+        )
+        # This is a simplified implementation of this operator just for tracing.
+        # Actual implementation should first promote the arguments
+        example_res = op(*real_sub_args).to(dtype=output_dtype)
+        example_value = deepcopy_to_fake_tensor(example_res, tx.fake_mode)
```

Comment:
As written it looks like we're constructing real Tensors and then running the Op on them. 

Ideally we would run FakeTensor propagation on the operator to get a FakeTensor output, and then call `wrap_fx_proxy` on that.